addons/account/tests/test_reconciliation_matching_rules.py

```diff
@@ -1045,74 +977,130 @@ def test_no_amount_check_exact_match(self):
             },
         })
 
-    def test_invoice_matching_rule_no_match_text_location_no_partner(self):
-        """ Test the behavior when no match text location and no match partner
-        are set in reconciliation model. It should match depending on either
-        the label, the note or the ref
-        """
-        self.bank_line_2.unlink()  # One line is enough for this test
-        self.bank_line_1.partner_id = None
+    @freeze_time('2019-01-01')
+    def test_invoice_matching_using_match_text_location(self):
+        @contextmanager
+        def rollback():
+            savepoint = self.cr.savepoint()
+            yield
+            savepoint.rollback()
 
-        self.rule_1.write({
-            'match_text_location_label': False,
-            'match_partner': False,
-            'match_partner_ids': [(5, 0, 0)],
-            'line_ids': [(5, 0, 0)],
+        rule = self._create_reconcile_model(
+            match_partner=False,
+            allow_payment_tolerance=False,
+            match_text_location_label=False,
+            match_text_location_reference=False,
+            match_text_location_note=False,
+        )
+        st_line = self._create_st_line(amount=1000)
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_date': '2019-01-01',
+            'invoice_line_ids': [Command.create({
+                'product_id': self.product_a.id,
+                'price_unit': 100,
+            })],
         })
+        invoice.action_post()
+        term_line = invoice.line_ids.filtered(lambda x: x.display_type == 'payment_term')
 
-        st_line_initial_vals = {'ref': None, 'payment_ref': 'nothing', 'narration': None}
-
-        self.bank_line_1.write(st_line_initial_vals)
-
-        # Nothing should match
-        self._check_statement_matching(self.rule_1, {
-            self.bank_line_1: {},
-        })
+        # No match at all.
+        self.assertDictEqual(
+            rule._apply_rules(st_line, None),
+            {},
+        )
 
-        for field in st_line_initial_vals:
-            # Test matching each field with the invoice ref
-            self.bank_line_1.write({**st_line_initial_vals, field: self.invoice_line_1.move_id.payment_reference})
+        with rollback():
+            term_line.name = "1234"
+            st_line.payment_ref = "1234"
 
-            self._check_statement_matching(self.rule_1, {
-                self.bank_line_1: {'amls': self.invoice_line_1, 'model': self.rule_1},
-            })
+            # Matching if no checkbox checked.
+            self.assertDictEqual(
+                rule._apply_rules(st_line, None),
+                {'amls': term_line, 'model': rule},
```

Comment:
I don't get why it should match here as amount is different in statement line and invoice line?